include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -226,25 +223,29 @@ destroy(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __
     typedef typename ::std::iterator_traits<_ForwardIterator>::value_type _ValueType;
     typedef typename ::std::iterator_traits<_ForwardIterator>::reference _ReferenceType;
 
-    const auto __is_parallel =
-        oneapi::dpl::__internal::__is_parallelization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec);
-    using _is_vector_type =
+    if constexpr (!::std::is_trivially_destructible_v<_ValueType>)
+    {
 #if (_PSTL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN || _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN)
-        ::std::conditional_t<
-            oneapi::dpl::__internal::__is_host_execution_policy<::std::decay_t<_ExecutionPolicy>>::value,
-            ::std::false_type,
-            decltype(oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(
-                __exec))>;
-#else
-        decltype(oneapi::dpl::__internal::__is_vectorization_preferred<_ExecutionPolicy, _ForwardIterator>(__exec));
+        if constexpr (oneapi::dpl::__internal::__is_host_execution_policy<::std::decay_t<_ExecutionPolicy>>::value)
+        {
+            constexpr auto __is_parallel =
+                oneapi::dpl::__internal::__is_parallelization_preferred<_ExecutionPolicy, _ForwardIterator>();
+
+            constexpr auto __is_vector = ::std::false_type{};
+
+            // TODO required to rewrite oneapi::dpl::__internal::__pattern_walk1 call by using tag
```

Comment:
@rarutyun how do you think, what is the right way to implement this workaround on tag dispatching?

Reply:
If I correctly understood the workaround I think that what we can do is:
- Instead of trying to decide if it's _vector_ or not right here we ask  `__select_backend` function for the correct tag, like in any other places
- Instead of passing the policy type as is we check if `SOMETHING_BROKEN` and if it's true we change the policy type by cutting _vectorized_ part. In other words:  
  - `parallel_unsequenced_policy` -> `parallel_policy`
  - `unsequenced_policy` -> `sequenced_policy`

And don't change the object itself, unless it's necessary, just the type that you pass to the `__select_backend`. Let me know if it works